pilot/pkg/proxy/envoy/v2/mem.go

```diff
@@ -160,6 +161,7 @@ func (sd *MemServiceDiscovery) AddInstance(service host.Name, instance *model.Se
 	key = fmt.Sprintf("%s:%s", service, instance.ServicePort.Name)
 	instanceList = sd.instancesByPortName[key]
 	sd.instancesByPortName[key] = append(instanceList, instance)
+	sd.mutex.Unlock()
```

Comment:
This is not right

Reply:
this is again for debugging - will remove

Reply:
ok

Reply:
btw, just for my knowledge, why this is not correct - instead of defer, I moved to explict unlock?

Reply:
There is a return in front, so leaves unlocked mutex

Reply:
I see. Did not notice the return. Any way I reverted.

Reply:
revert